src/reducers/7.12/reducer.ts

```diff
@@ -298,10 +298,18 @@ interface DeletePEMGenerationForAcceptorAction
 
 interface AddAcceptorAction extends ArtemisReducerActionBase {
   operation: ArtemisReducerOperations.addAcceptor;
+  payload: {
+    name: string;
+    port: number;
+  };
 }
 
 interface AddConnectorAction extends ArtemisReducerActionBase {
   operation: ArtemisReducerOperations.addConnector;
+  payload: {
+    name: string;
+    port: number;
+  };
```

Comment:
By asking the calling code to provide the `name` and `port` to the Add connector action, we're externalizing the logic of producing a unique port number to the UI code. That's making us diverging from the primary goal of the reducer: centralizing the update logic.

What we want to have instead is a addConnector function able to search for a free port.